### PR TITLE
[SQL] Parse model schema string from python

### PR DESCRIPTION
**Pull request by eddyxu**
Parse model schema string from python. It uses the same antlr gramma from #115

### CODE REVIEW

**Comment by eddyxu** [comment]
This PR does not support UDT yet

File: python/rikai/spark/sql/generated/RikaiModelSchema.interp [hunk 1]
```diff
@@ -0,0 +1,33 @@
+token literal names:
+null
+'<'
+','
+'>'
+':'
+'struct'
+'array'
+null
```
Comment (changhiskhan) [inline python/rikai/spark/sql/generated/RikaiModelSchema.interp:9]: why are there so many `null`s?

File: python/rikai/spark/sql/schema.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        return StructField(name, dataType)
+
+    def visitArrayType(
+        self, ctx: RikaiModelSchemaParser.ArrayTypeContext
+    ) -> ArrayType:
+        return ArrayType(self.visit(ctx.fieldType()))
+
+    def visitUnquotedIdentifier(
+        self, ctx: RikaiModelSchemaParser.UnquotedIdentifierContext
+    ):
```
Comment (changhiskhan) [inline python/rikai/spark/sql/schema.py:87]: nit: return type?

File: python/rikai/spark/sql/schema.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+def parse_schema(schema_str: str) -> DataType:
+    input_stream = InputStream(schema_str)
+    lexer = RikaiModelSchemaLexer(input_stream)
+    stream = CommonTokenStream(lexer)
+    parser = RikaiModelSchemaParser(stream)
+
+    visitor = SchemaBuilder()
+    schema = visitor.visit(parser.schema())
+    # TODO: we should add error listener to Antlr Parser.
```
Comment (changhiskhan) [inline python/rikai/spark/sql/schema.py:114]: create a gh issue?

**Review by changhiskhan — APPROVED** [review]
only found nits. 

how does the joke go? give a reviewer 10 lines and he'll come up with 10 things to say. give a reviewer 1000 lines and he'll say "looks good" everytime lol

File: python/rikai/spark/sql/generated/RikaiModelSchema.interp (hunk 1 above)
Comment (eddyxu) [reply python/rikai/spark/sql/generated/RikaiModelSchema.interp:9]: These are auto-generated from antlr :)

**Comment by eddyxu** [comment]
Lol 900 lines of these are autogenarated from `antlr` tho.

File: python/rikai/spark/sql/schema.py (hunk 3 above)
Comment (eddyxu) [reply python/rikai/spark/sql/schema.py:114]: 👌

File: python/rikai/spark/sql/schema.py (hunk 2 above)
Comment (eddyxu) [reply python/rikai/spark/sql/schema.py:87]: done

**Pull request merged**